go/internal/feast/registry.go

```diff
@@ -228,89 +248,113 @@ func (r *Registry) listOnDemandFeatureViews(project string) []*OnDemandFeatureVi
 	Returns empty list if project not found
 */
 
-func (r *Registry) listRequestFeatureViews(project string) []*RequestFeatureView {
+func (r *Registry) listRequestFeatureViews(project string, allowCache bool) ([]*RequestFeatureView, error) {
+	_, err := r.getRegistryProto(allowCache)
+	if err != nil {
+		return nil, err
+	}
 	if requestFeatureViewProtos, ok := r.cachedRequestFeatureViews[project]; !ok {
-		return []*RequestFeatureView{}
+		return []*RequestFeatureView{}, nil
 	} else {
 		requestFeatureViews := make([]*RequestFeatureView, len(requestFeatureViewProtos))
 		index := 0
 		for _, requestFeatureViewProto := range requestFeatureViewProtos {
 			requestFeatureViews[index] = NewRequestFeatureViewFromProto(requestFeatureViewProto)
 			index += 1
 		}
-		return requestFeatureViews
+		return requestFeatureViews, nil
 	}
 }
 
-func (r *Registry) getEntity(project, entityName string) (*Entity, error) {
+func (r *Registry) getEntity(project, entityName string, allowCache bool) (*Entity, error) {
+	_, err := r.getRegistryProto(allowCache)
+	if err != nil {
+		return nil, err
+	}
 	if entities, ok := r.cachedEntities[project]; !ok {
-		return nil, errors.New(fmt.Sprintf("project %s not found in getEntity", project))
+		return nil, fmt.Errorf("project %s not found in getEntity", project)
```

Comment:
nit: if we keep this `cachedEntities` approach, this error message isn't super descriptive

maybe something like `no cached entities found for project %s`?